### PR TITLE
xfstest: Use namespaces instead of partitions

### PR DESCRIPTION
**Pull request by kvaneesh**
Using partition with nvdimm is not really recommended.

Signed-off-by: Aneesh Kumar K.V <aneesh.kumar@linux.ibm.com>

### CODE REVIEW

**Comment by kvaneesh** [comment]
@harish-24  please review

File: fs/xfstests.py [hunk 1]
```diff
@@ -40,6 +42,79 @@ class Xfstests(Test):
...
+        pmem_dev = self.plib.run_ndctl_list_val(namespaces[1], 'blockdev')
+        self.scratch_dev = "/dev/%s" % pmem_dev
+        self.devices.extend([self.test_dev, self.scratch_dev])
+
+        if logflag:
+            # log device to be created in sector mode
+            self.plib.destroy_namespace(region=region_ldev, force=True)
+            namespace_size = self.get_half_region_size(region=region_ldev)
+            self.plib.create_namespace(region=region_ldev, mode='sector',
+                                       sector_size='512', size=namespace_size)
```
Comment (harish-24) [inline fs/xfstests.py]: Logdev for xfs filesystem should be less than 2G. 
```
# mkfs.xfs -f -b size=64k -s size=512 -m reflink=0 -l logdev=/dev/pmem1s /dev/pmem0
log size 53270740992 bytes too large, maximum size is 2136997888 bytes
...
```

Comment (kvaneesh) [reply fs/xfstests.py]: updated with 
```
+            namespace_size = self.get_half_region_size(region=region_ldev)
+            # XFS restrict max log size to 2136997888
+            namespace_size = max(namespace_size, 2136997888)

```

Comment (harish-24) [reply fs/xfstests.py]: It should be `min(namespace_size, 2136997888)`

File: fs/xfstests.py [hunk 2]
```diff
@@ -40,6 +42,81 @@ class Xfstests(Test):
...
+            self.cancel("Nvdimm test with log flag need two region support")
+
+        regions = sorted(self.plib.run_ndctl_list('-R'),
+                         key=lambda i: i['size'], reverse=True)
+        region = self.plib.run_ndctl_list_val(regions[0], 'dev')
+        if self.plib.is_region_legacy(region):
+            self.cancel("Not supported with legacy regions")
+
+        if logflag:
+            region_ldev = self.plib.run_ndctl_list_val(regions[1], 'dev')
```
Comment (harish-24) [inline fs/xfstests.py]: this would fail if there is only one region.

File: fs/xfstests.py [hunk 3]
```diff
@@ -40,6 +42,81 @@ class Xfstests(Test):
...
+        pmem_dev = self.plib.run_ndctl_list_val(namespaces[1], 'blockdev')
+        self.scratch_dev = "/dev/%s" % pmem_dev
+        self.devices.extend([self.test_dev, self.scratch_dev])
+
+        if logflag:
+            # log device to be created in sector mode
+            self.plib.destroy_namespace(region=region_ldev, force=True)
+            namespace_size = self.get_half_region_size(region=region_ldev)
+            # XFS restrict max log size to 2136997888
+            namespace_size = max(namespace_size, 2136997888)
```
Comment (harish-24) [inline fs/xfstests.py]: It should be min(namespace_size, 2136997888)

File: fs/xfstests.py [hunk 4]
```diff
@@ -40,6 +42,81 @@ class Xfstests(Test):
...
+        regions = sorted(self.plib.run_ndctl_list('-R'),
+                         key=lambda i: i['size'], reverse=True)
+        if not regions:
+            self.cancel("Nvdimm test with no region support")
+
+        if logflag and not len(regions) > 1:
+            self.cancel("Nvdimm test with log flag need two region support")
+
+        regions = sorted(self.plib.run_ndctl_list('-R'),
+                         key=lambda i: i['size'], reverse=True)
```
Comment (harish-24) [inline fs/xfstests.py]: This is repeating again, so we can remove it.

**Pull request merged**